src/main/scala/scala/concurrent/java8/FutureConvertersImpl.scala

```diff
@@ -66,8 +66,15 @@ object FuturesConvertersImpl {
       cf
     }
 
-    override def toCompletableFuture(): CompletableFuture[T] =
-      throw new UnsupportedOperationException("this CompletionStage represents a read-only Scala Future")
+    /**
+     * @inheritdoc
+     *
+     * WARNING: completing the result of this method will not complete the underlying
```

Comment:
> There is no underlying Scala Future or Promise for this class? (CF)

The underlying future is the one passed to `toJava`:

```
  def toJava[T](f: Future[T]): CompletionStage[T] = {
    val cf = new CF[T]
    implicit val ec = InternalCallbackExecutor
    f onComplete cf
    cf
  }
```

In my other PR, that also becomes a field of `CF` so that we can efficiently unwrap it.